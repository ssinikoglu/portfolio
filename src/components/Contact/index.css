.contactForm__title {
  font-size: 2rem;
}

.contactForm__label {
  font-size: 1.25rem;
}

.contactForm__button {
  font-size: 1.1rem;
}
