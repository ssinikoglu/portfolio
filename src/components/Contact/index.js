import React, {useState} from "react";
import {
  MDBMask,
  MDBContainer,
  MDBRow,
  MDBCol,
  MDBBtn,
  MDBIcon
} from "mdbreact";
import { ReactComponent as ContactImage } from "../../images/contact.svg";
import "./index.css";

// Converts object to proper form data(string)
const encodeFormData = (data) => {
  return Object.keys(data)
    .map((key) => encodeURIComponent(key) + "=" + encodeURIComponent(data[key]))
    .join("&");
};

const Contact = () => {
  const [formValue, setFormValue] = useState({
    name: "",
    email: "",
    subject: "",
    message: "",
  });

  const handleChange = (e) => setFormValue({ [e.target.name]: e.target.value });

  const handleSubmit = (e) => {
    e.preventDefault();
    fetch("/", {
      method: "POST",
      "data-netlify": "true",
      headers: { "Content-Type": "application/x-www-form-urlencoded" },
      body: encodeFormData({ "form-name": "contact", ...formValue }),
    })
      .then(() => {
        alert("Thanks for messaging!");
        setFormValue({
          name: "",
          email: "",
          subject: "",
          message: "",
        });
      })
      .catch((error) => alert(error));
  };

  return (
    <>
      <MDBContainer>
        <MDBRow>
          <MDBMask
            overlay="black"
            className="flex-center flex-row black-text text-left"
          >
            <MDBCol>
              <MDBCol md="10" className="ml-5">
                <form className="needs-validation" onSubmit={handleSubmit}>
                  <h2 className="h4 text-center mb-4 contactForm__title">
                    <strong>
                      <b>Keep in touch</b>
                    </strong>
                  </h2>
                  <label
                    htmlFor="validationCustom01"
                    className="black-text font-weight-bold contactForm__label"
                  >
                    Name:
                  </label>
                  <input
                    type="text"
                    id="validationCustom01"
                    className="form-control"
                    name="name"
                    value={formValue.name}
                    onChange={handleChange}
                    required
                  />
                  <br />
                  <label
                    htmlFor="defaultFormContactEmailEx"
                    className="black-text font-weight-bold contactForm__label"
                  >
                    E-mail:
                  </label>
                  <input
                    type="email"
                    id="defaultFormContactEmailEx"
                    className="form-control"
                    value={formValue.email}
                    onChange={handleChange}
                    name="email"
                    required
                  />
                  <br />
                  <label
                    htmlFor="defaultFormContactSubjectEx"
                    className="black-text font-weight-bold contactForm__label"
                  >
                    Subject:
                  </label>
                  <input
                    type="text"
                    id="defaultFormContactSubjectEx"
                    className="form-control"
                    value={formValue.subject}
                    onChange={handleChange}
                    name="subject"
                  />
                  <br />
                  <label
                    htmlFor="defaultFormContactMessageEx"
                    className="black-text font-weight-bold contactForm__label"
                  >
                    Message:
                  </label>
                  <textarea
                    type="text"
                    id="defaultFormContactMessageEx"
                    className="form-control"
                    rows="3"
                    name="message"
                    value={formValue.message}
                    onChange={handleChange}
                    required
                  />
                  <div className="text-center mt-4">
                    <MDBBtn
                      color="black font-weight-bold"
                      outline
                      type="submit"
                      className="btn-block waves-effect contactForm__button"
                      aria-hidden="true"
                    >
                      Send
                      <MDBIcon far icon="paper-plane" className="ml-2" />
                    </MDBBtn>
                  </div>
                </form>
              </MDBCol>
            </MDBCol>
            <MDBCol>
              <ContactImage
                className="img-fluid mb-5"
                style={{
                  float: "right",
                  width: "100%",
                }}
              />
            </MDBCol>
          </MDBMask>
        </MDBRow>
      </MDBContainer>
    </>
  );
};

export default Contact;
