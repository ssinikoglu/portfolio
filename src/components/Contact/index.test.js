import React from "react";
import Contact from "./index.js";
import renderer from "react-test-renderer";

it("Contact snapshot test ", () => {
  const tree = renderer.create(<Contact />).toJSON();
  expect(tree).toMatchSnapshot();
});
