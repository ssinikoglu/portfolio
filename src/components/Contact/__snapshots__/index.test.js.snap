// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact snapshot test  1`] = `
<div
  className="container"
  data-test="container"
>
  <div
    className="row"
    data-test="row"
  >
    <div
      className="mask rgba-black flex-center flex-row black-text text-left"
      data-test="mask"
    >
      <div
        className="col"
        data-test="col"
      >
        <div
          className="col-md-10 ml-5"
          data-test="col"
        >
          <form
            className="needs-validation"
            data-netlify="true"
            data-netlify-honeypot="bot-field"
            method="post"
            name="contact"
            onSubmit={[Function]}
          >
            <input
              name="form-name"
              type="hidden"
              value="contact"
            />
            <h2
              className="h4 text-center mb-4 contactForm__title"
            >
              <strong>
                <b>
                  Keep in touch
                </b>
              </strong>
            </h2>
            <label
              className="black-text font-weight-bold contactForm__label"
              htmlFor="validationCustom01"
            >
              Name:
            </label>
            <input
              className="form-control"
              id="validationCustom01"
              name="name"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <br />
            <label
              className="black-text font-weight-bold contactForm__label"
              htmlFor="defaultFormContactEmailEx"
            >
              E-mail:
            </label>
            <input
              className="form-control"
              id="defaultFormContactEmailEx"
              name="email"
              onChange={[Function]}
              required={true}
              type="email"
              value=""
            />
            <br />
            <label
              className="black-text font-weight-bold contactForm__label"
              htmlFor="defaultFormContactSubjectEx"
            >
              Subject:
            </label>
            <input
              className="form-control"
              id="defaultFormContactSubjectEx"
              name="subject"
              onChange={[Function]}
              type="text"
              value=""
            />
            <br />
            <label
              className="black-text font-weight-bold contactForm__label"
              htmlFor="defaultFormContactMessageEx"
            >
              Message:
            </label>
            <textarea
              className="form-control mb-1"
              id="defaultFormContactMessageEx"
              name="message"
              onChange={[Function]}
              required={true}
              rows="3"
              type="text"
              value=""
            />
            <div
              className="text-center mt-4"
            >
              <button
                aria-hidden="true"
                className="btn-black font-weight-bold btn-outline-black font-weight-bold btn Ripple-parent btn-block waves-effect contactForm__button"
                data-test="button"
                onMouseUp={[Function]}
                onTouchStart={[Function]}
                type="submit"
              >
                Send
                <i
                  className="far fa-paper-plane ml-2"
                  data-test="fa"
                />
                <div
                  className="Ripple Ripple-outline "
                  data-test="waves"
                  style={
                    Object {
                      "height": "0px",
                      "left": "0px",
                      "top": "0px",
                      "width": "0px",
                    }
                  }
                />
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        className="col"
        data-test="col"
      >
        <svg
          className="img-fluid mb-5"
          style={
            Object {
              "float": "right",
              "width": "100%",
            }
          }
        >
          contact.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;
